### ISSUE TITLE
Model recognition of "proximity to parallel"

### ISSUE THREAD

**Issue by BLFiedler** [phetsims/quadrilateral#68]
I am creating this issue for archival purposes, as the idea for it has been brought up numerous times. As of February 2022, this feature is not in development.

The idea of recognizing when a user has moved the vertices in a way that puts them "close to" a parallelogram shape has been brought up several times in terms of:
1. Feedback based on proximity, e.g., sound or haptics that play/trigger when a vertex reaches a position relative to the other vertices
2. Snapping behavior when a mouse cursor release or touch release occurs, or a keyboard movement that would otherwise jump the "parallelogram" state (as occurs with Ratio and Proportions hand positions relative to the "fitness level")

There are a few obstacles to implementing this model behavior that I feel would require excessive developer time and troubleshooting to not make it worth an effort when there are other priorities and features to expand the scaffolding of the exploration of the shape(s).

TL;DR - I believe TOO MANY THINGS CHANGE at once to make implementation sensical, mappable, and worthwhile.

With respect to 1:
- Decisions with respect to what a "perfect parallelogram" is when the angles of one corner are dependent on adjacent corners is not straightforward. This is made more difficult when more than one vertex is moving at once.
- Similar to feedback we've received on sounds and Voicing, especially when more than one vertex is changing, it is not clear what type of feedback could be provided that indicates to a user WHAT is changing or what the feedback about proximity is telling them. If a sound place while I move two vertices, what vertex is the sim telling me that I need to move? And in what direction?
   - We have a rough instantiation of this behavior with the early sonification schemes that map sounds to the lengths and angle changes. It was not clear how to react or use the sounds to achieve a parallelogram.

With respect to 2:
- Snapping behavior again introduces the question of what a "perfect parallelogram" is. In a 2 dimensional space, what variables snap? The X position of the vertex? Y? Both? How? If the other vertices are not also perfect, do they snap as well? To what extent does the shape change/adapt to become a perfect shape that removes control or alters the state of the shape the learner created? And how are they informed if other aspects of the shape need to change?
- These are a lot of questions to solve and each is a point of troubleshooting, and are compounded when we consider multiple vertices being picked up and released at different times.

Tagging @jessegreenberg and @terracoda for awareness of this issue, but closing. Notes will be copied into the Project history document.
